client/banner.js

```diff
@@ -10,17 +12,77 @@ import jQuery from 'jquery';
 import '../assets/stylesheets/banner.scss';
 
 jQuery( document ).ready( ( $ ) => {
-	$( '.wcc-admin-notice .wcc-install-jetpack' ).one( 'click', function() {
+	$( '.woocommerce-services__connect-jetpack' ).one( 'click', function( event ) {
+		event.preventDefault();
 		const btn = $( this );
+		btn.addClass( 'disabled' );
+
+		installStep()
+			.then( activateStep )
+			.then( connectStep )
+			.fail( function( error ) {
+				let errorMessage = error;
+				if ( ! error ) {
+					errorMessage = wcs_nux_notice.translations.defaultError;
+				}
+				if ( error && error.install && 'plugin' === error.install ) {
+					// plugin install error
+					errorMessage = wcs_nux_notice.translations.installError;
+				}
+				$( '<p/>', {
+					// eslint-disable-next-line quote-props
+					'class': 'woocommerce-services__jetpack-install-error-message',
+					text: errorMessage,
+				} ).insertAfter( btn );
+				btn.remove();
+			} );
+
+		function installStep() {
+			if ( 'uninstalled' === wcs_nux_notice.initial_install_status ) {
+				return $.when()
+					.then( function() {
+						btn.html( wp.updates.l10n.installing );
+						return wp.updates.installPlugin( { slug: 'jetpack' } );
+					} );
+			}
+			return $.Deferred().resolve();
+		}
 
-		btn.html( wp.updates.l10n.installing )
-			.addClass( 'disabled' );
+		function activateStep() {
+			if ( 'installed' === wcs_nux_notice.initial_install_status ||
+				'uninstalled' === wcs_nux_notice.initial_install_status
+			) {
+				return $.when()
+				.then( function() {
+					btn.html( wcs_nux_notice.translations.activating );
+					return $.post( ajaxurl, {
+						action: 'woocommerce_services_activate_jetpack',
```

Comment:
@v18 you'd install using the `Plugin_Upgrader` class, see this example: https://github.com/Automattic/jetpack/blob/master/3rd-party/woocommerce-services.php#L101